source/jormungandr/tests/direct_path_asgard_integration_tests.py

```diff
@@ -0,0 +1,162 @@
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+from __future__ import absolute_import
+from .tests_mechanism import AbstractTestFixture, dataset
+from jormungandr.street_network.asgard import Asgard
+import logging
+from navitiacommon import response_pb2, type_pb2
+
+MOCKED_ASGARD_CONF = [
+    {
+        "modes": ['walking', 'car', 'bss', 'bike'],
+        "class": "tests.direct_path_asgard_integration_tests.MockAsgard",
+        "args": {
+            "costing_options": {"bicycle": {"bicycle_type": "Hybrid"}},
+            "api_key": "",
+            "asgard_socket": "bob_socket",
+            "service_url": "http://bob.com",
+            "timeout": 10,
+        },
+    }
+]
+s_coord = "0.0000898312;0.0000898312"  # coordinate of S in the dataset
+r_coord = "0.00188646;0.00071865"  # coordinate of R in the dataset
+
+journey_basic_query = "journeys?from={from_coord}&to={to_coord}&datetime={datetime}".format(
+    from_coord=s_coord, to_coord=r_coord, datetime="20120614T080000"
+)
+
+
+def route_response(mode):
+    map_mode_dist = {"walking": 200, "car": 50, "bike": 100}
+    map_mode_time = {"walking": 2000, "car": 500, "bike": 1000}
+    response = response_pb2.Response()
+    response.response_type = response_pb2.ITINERARY_FOUND
+    journey = response.journeys.add()
+    journey.nb_transfers = 0
+    journey.nb_sections = 1
+    journey.departure_date_time = 1548669936
+    journey.arrival_date_time = 1548670472
+    journey.requested_date_time = 1548669936
+
+    duration = map_mode_time.get(mode)
+    distance = map_mode_dist.get(mode)
+    journey.duration = duration
+    journey.durations.total = duration
+    section = journey.sections.add()
+    if mode == "walking":
+        journey.durations.walking = duration
+        journey.distances.walking = distance
+        section.street_network.mode = response_pb2.Walking
+    elif mode == "car":
+        journey.durations.car = duration
+        journey.distances.car = distance
+        section.street_network.mode = response_pb2.Car
+    else:
+        journey.durations.bike = duration
+        journey.distances.bike = distance
+        section.street_network.mode = response_pb2.Bike
+
+    section.type = response_pb2.STREET_NETWORK
+    section.duration = duration
+    section.length = distance
+    section.id = "section"
+    section.begin_date_time = 1548669936
+    section.end_date_time = 1548670472
+
+    return response
+
+
+def response_valid(request):
+    if request.requested_api == type_pb2.direct_path:
+        return route_response(request.direct_path.streetnetwork_params.origin_mode)
+    else:
+        return response_pb2.Response()
+
+
+def check_journeys(resp):
+    assert not resp.get('journeys') or sum([1 for j in resp['journeys'] if j['type'] == "best"]) == 1
+
+
+class MockAsgard(Asgard):
+    def __init__(
```

Comment:
```suggestion
    def __init__(*args, **kwargs):
        super(self).__init__(*args, **kwargs)
```

Reply:
Doesn't work because of the interface. Thank you anyway ! :D